src/cli.zig

```diff
@@ -1,78 +1,137 @@
         sub: SubInfo,
 
         pub const Cmd = S;
         pub const SubInfo = computeCmdAndOptBasicInfo(S, null).SubInfo;
     };
 }
 
-/// Returns a struct with all the same fields as input struct `S`,
-/// but where each field is of type `OptionInfo(FieldType)`.
-pub fn OptionInfoGroup(comptime S: type) type {
-    const Type = std.builtin.Type;
-    const s_info = @typeInfo(S).Struct;
-
-    var sub_fields: [s_info.fields.len]Type.StructField = undefined;
-    for (&sub_fields, s_info.fields) |*new_s_field, s_field| {
-        if (@typeInfo(s_field.type) == .Union or
-            (@typeInfo(s_field.type) == .Optional and
-            @typeInfo(@typeInfo(s_field.type).Optional.child) == .Union))
-        {
-            @compileError("The subcommand field cannot be part of an option group");
-        }
-
-        new_s_field.* = .{
-            .name = s_field.name,
-            .type = OptionInfo(s_field.type),
-            .default_value = null,
-            .is_comptime = false,
-            .alignment = 0,
-        };
-    }
-
-    return @Type(.{ .Struct = .{
-        .layout = .auto,
-        .backing_integer = null,
-        .fields = &sub_fields,
-        .is_tuple = false,
-        .decls = &.{},
-    } });
-}
-
-/// Describes how the value for a `[]const u8` should be interpreted.
-pub const BytesConfig = enum {
-    /// Should be parsed as a single string value.
-    string,
-    /// Should be parsed as a list of unsigned 8 bit integers.
-    list,
-
-    /// Used in place of `BytesConfig` when `[]const u8` is in an optional or list,
-    /// since the only logical thing for such a composite to be would be a string.
-    pub const StringOnly = enum { string };
+pub const CommandHelp = struct {
+    /// Brief description of the command to be displayed as part of the parent command's help message,
+    /// recommended but not required to be one line.
+    ///
+    /// Also displayed as part of the target command's help message, just before the long description.
+    short: []const u8,
+    /// Long description of the command to be displayed as part of the target command's help message,
+    /// optionally applicable if a more verbose description is helpful or required.
+    ///
+    /// It is recommended that this be worded in a way which is not redundant with the brief, since it
+    /// will be displayed as a paragraph that follows the brief.
+    long: ?[]const u8,
 };
 
+/// This struct describes everything related to a given option within the scope of a (sub)command.
+///
+/// Similar to `CommandInfo`, it can be instantiated directly at the site of usage, or it can
+/// be pre-declared, which allows re-using the option info in multiple sub-commands.
 pub fn OptionInfo(comptime Opt: type) type {
     return struct {
+        /// A named option will be specifiable as `--{name_override orelse field_name}(=<value>)?`,
+        /// irrespective of position within the scope of the (sub)command.
+        /// A positional option
```

Comment:
Missing the rest of this sentence?

Reply:
e4fc4a41f879358a986caa635b8ad9883d1961bb